guppylang-internals/src/guppylang_internals/tys/qubit.py

```diff
@@ -0,0 +1,61 @@
+import functools
+from typing import Any, cast
+
+from guppylang_internals.definition.ty import TypeDef
+from guppylang_internals.tys.arg import TypeArg
+from guppylang_internals.tys.common import Visitor
+from guppylang_internals.tys.ty import OpaqueType, Type
+
+
+def qubit_ty() -> Type:
```

Comment:
Consider using https://docs.python.org/3/library/functools.html#functools.cache to memoise this function